### PR TITLE
rn: refactor aspect ratio and reduced UI detectors

### PR DESCRIPTION
**Pull request by saghul**

### CODE REVIEW

**Comment by saghul** [comment]
Marked a draft, since it needs some testing.

**Comment by saghul** [comment]
Yay cirtular references. Hold on...

File: react/features/base/responsive-ui/middleware.native.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+ * @param {Store} store - The redux store.
+ * @private
+ * @returns {void}
+ */
+function _onDimensionsChange(dimensions, store) {
+    const { width, height } = dimensions.window;
+    const { dispatch } = store;
+
+    dispatch(setAspectRatio(width, height));
+    dispatch(setReducedUI(width, height));
```
Comment (virtuacoplenny) [inline react/features/base/responsive-ui/middleware.native.js]: Do setAspectRatio and setReducedUI always get called together with the same values? Is it worth combining into one action, assuming what they do are very related?

Comment (saghul) [reply react/features/base/responsive-ui/middleware.native.js]: They do use the same values, but they compute different things so I thought it made sense to keep them separate. I don’t event know what name I’d give that thing... WDTY?

File: react/features/base/responsive-ui/actions.js [hunk 2]
```diff
@@ -12,9 +12,7 @@ import type { Dispatch } from 'redux';
  * very brittle because it's completely disconnected from the UI which wants to
  * be rendered and, naturally, it broke on iPad where even the secondary Toolbar
  * didn't fit in the height. We do need to measure the actual UI at runtime and
- * determine whether and how to render it. I'm bumping from 240 to 300 because I
- * don't have the time now to refactor {@code ReducedUIDetector} or rip it out
- * completely.
+ * determine whether and how to render it. I'm bumping from 240 to 300.
```
Comment (zbettenbuk) [inline react/features/base/responsive-ui/actions.js]: Do we still need the "I'm bumping..." part?

File: react/features/base/responsive-ui/middleware.native.js (hunk 1 above)
Comment (zbettenbuk) [reply react/features/base/responsive-ui/middleware.native.js]: What i'd do is to only dispatch dimensions change event here, and let the appropriate features handle them, meaning the reduced UI should listen to dimension change, not the dimension detector inform the reduced UI to do something.

File: react/features/base/responsive-ui/actions.js (hunk 2 above)
Comment (saghul) [reply react/features/base/responsive-ui/actions.js]: Nah, I can axe that.

File: react/features/base/responsive-ui/middleware.native.js (hunk 1 above)
Comment (saghul) [reply react/features/base/responsive-ui/middleware.native.js]: All those 3 things are in the same "umbrella" feature, this one. Dispatching a dimensions changed just to catch it on the same middleware to then do these 2 things felt unnecessary so I took this shortcut. If any other entity ever needs to be aware of the specific dimensions (I hope not, that would be kinda bad) we can change it.

**Pull request merged**